### PR TITLE
Skip tests/dockerfiles per branch

### PR DESCRIPTION
**Pull request by skonto**
- We have a case where we don't want to run perf tests for the `release-next` branch but we want to have the targets and the dockerfiles available so when a new release branch is cut to use them (they are under ./openshift dir by convention and they affect release-next).
- Tested in unit test. Also tested with the setting:
 ```
    "release-v1.11":
      openShiftVersions:
        - version: 4.14
        - version: 4.11
      skipE2EMatches:
        - "perf-tests$"
      skipDockerFilesMatches:
        - "openshift/ci-operator/knative-perf-images.*"
      openShiftVersions:
        - version: 4.14
        - version: 4.11
```
The above setting generates the following git diff:
```
	modified:   ci-operator/config/openshift-knative/serving/openshift-knative-serving-release-v1.11__411.yaml
	modified:   ci-operator/config/openshift-knative/serving/openshift-knative-serving-release-v1.11__414.yaml
	deleted:    ci-operator/config/openshift-knative/serving/openshift-knative-serving-release-v1.8__410.yaml
	deleted:    ci-operator/config/openshift-knative/serving/openshift-knative-serving-release-v1.8__413.yaml
	modified:   ci-operator/jobs/openshift-knative/serving/openshift-knative-serving-release-v1.11-periodics.yaml
	modified:   ci-operator/jobs/openshift-knative/serving/openshift-knative-serving-release-v1.11-presubmits.yaml
	deleted:    ci-operator/jobs/openshift-knative/serving/openshift-knative-serving-release-v1.8-periodics.yaml
	deleted:    ci-operator/jobs/openshift-knative/serving/openshift-knative-serving-release-v1.8-postsubmits.yaml
	deleted:    ci-operator/jobs/openshift-knative/serving/openshift-knative-serving-release-v1.8-presubmits.yaml
	modified:   core-services/image-mirroring/knative/mapping_knative_knative-v1.11_serving_quay
	deleted:    core-services/image-mirroring/knative/mapping_knative_knative-v1.8_serving_quay
```
Actual diff contents [here](https://gist.githubusercontent.com/skonto/13d149cb6e3a5b46aeb288baf47a4f0b/raw/7ddfaba9d7d122539ca743487ff5e9881f1cc975/gistfile1.txt).

### CODE REVIEW

File: config/serving.yaml [hunk 1]
```diff
@@ -19,6 +19,10 @@ config:
         - version: 4.14
         - version: 4.11
     "release-next":
+      skipE2EMatches:
```
Comment (skonto) [inline config/serving.yaml:22]: At some point if we add more rules then the config model might have to be revised.

Comment (ReToCode) [reply config/serving.yaml:22]: +1 on this, but its a good thing that we slowly find all differences

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

**Pull request merged**